From 56ca740bbc6c004d2e38c2d9f549013f979da470 Mon Sep 17 00:00:00 2001
From: Weijie Gao <weijie.gao@mediatek.com>
Date: Thu, 21 Aug 2025 10:35:02 +0800
Subject: [PATCH 07/23] board: mediatek: fix NULL upgrade handler of nor+emmc
 bootmenu

Remove unexpected unused macro for bl upgrading

Change-Id: I7c3af378259e90e52afb43cd62501e4046c49e73
Signed-off-by: Weijie Gao <weijie.gao@mediatek.com>
CR-Id: WCNCR00180092
---
 board/mediatek/common/bootmenu_snor_emmc.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/board/mediatek/common/bootmenu_snor_emmc.c b/board/mediatek/common/bootmenu_snor_emmc.c
index 3b853aea7b3..fe97ebac37d 100644
--- a/board/mediatek/common/bootmenu_snor_emmc.c
+++ b/board/mediatek/common/bootmenu_snor_emmc.c
@@ -22,11 +22,7 @@ static const struct data_part_entry snor_emmc_parts[] = {
 		.abbr = "fip",
 		.env_name = "bootfile.fip",
 		.validate = generic_validate_fip,
-#if defined(CONFIG_FIP_IN_SPI_NOR)
 		.write = generic_mtd_write_fip,
-#elif defined(CONFIG_FIP_IN_EMMC)
-		.write = generic_mmc_write_fip,
-#endif
 		.post_action = UPGRADE_ACTION_CUSTOM,
 		.do_post_action = generic_invalidate_env,
 	},
@@ -36,11 +32,7 @@ static const struct data_part_entry snor_emmc_parts[] = {
 		.abbr = "bl31",
 		.env_name = "bootfile.bl31",
 		.validate = generic_validate_bl31,
-#if defined(CONFIG_FIP_IN_SPI_NOR)
 		.write = generic_mtd_update_bl31,
-#elif defined(CONFIG_FIP_IN_EMMC)
-		.write = generic_mmc_update_bl31,
-#endif
 		.post_action = UPGRADE_ACTION_CUSTOM,
 	},
 	{
@@ -48,11 +40,7 @@ static const struct data_part_entry snor_emmc_parts[] = {
 		.abbr = "bl33",
 		.env_name = "bootfile.bl33",
 		.validate = generic_validate_bl33,
-#if defined(CONFIG_FIP_IN_SPI_NOR)
 		.write = generic_mtd_update_bl33,
-#elif defined(CONFIG_FIP_IN_EMMC)
-		.write = generic_mmc_update_bl33,
-#endif
 		.post_action = UPGRADE_ACTION_CUSTOM,
 		.do_post_action = generic_invalidate_env,
 	},
-- 
2.45.2

